js/modules/networkgraph/node-animation.js

```diff
@@ -0,0 +1,150 @@
+                // Get width and height if pure text nodes (stack labels)
+                if (!label.width) {
+                    bBox = label.getBBox();
+                    label.width = bBox.width;
+                    label.height = bBox.height;
+                    // Labels positions are computed from top left corner,
+                    // so we need to substract the text
+                    // height from text nodes too.
+                    lineHeightCorrection = ren
+                        .fontMetrics(null, label.element).h;
+                }
+                return {
+                    x: pos.x + (parent.translateX || 0) + padding,
+                    y: pos.y + (parent.translateY || 0) + padding -
+                        lineHeightCorrection,
+                    width: label.width - 2 * padding,
+                    height: label.height - 2 * padding
+                };
+            }
+        };
+        for (i = 0; i < len; i++) {
+            label = labels[i];
+            if (label) {
+                // Mark with initial opacity
+                label.oldOpacity = label.opacity;
+                label.newOpacity = 1;
+                label.absoluteBox = getAbsoluteBox(label);
+            }
+        }
+        // Prevent a situation in a gradually rising slope, that each label will
+        // hide the previous one because the previous one always has lower rank.
+        labels.sort(function (a, b) {
+            return (b.labelrank || 0) - (a.labelrank || 0);
+        });
+        // Detect overlapping labels
+        for (i = 0; i < len; i++) {
+            label1 = labels[i];
+            box1 = label1 && label1.absoluteBox;
+            for (j = i + 1; j < len; ++j) {
+                label2 = labels[j];
+                box2 = label2 && label2.absoluteBox;
+                if (box1 &&
+                    box2 &&
+                    label1 !== label2 && // #6465, polar chart with connectEnds
+                    label1.newOpacity !== 0 &&
+                    label2.newOpacity !== 0) {
+                    if (isIntersectRect(box1, box2)) {
+                        (label1.labelrank < label2.labelrank ? label1 : label2)
+                            .newOpacity = 0;
+                    }
+                }
+            }
+        }
+        var isLabelAffected = false;
+        // Hide or show
+        labels.forEach(function (label) {
+            var complete, newOpacity;
+            if (label) {
+                newOpacity = label.newOpacity;
+                if (label.oldOpacity !== newOpacity) {
+                    // Make sure the label is completely hidden
+                    // to avoid catching clicks (#4362)
+                    if (label.alignAttr && label.placed) { // data labels
+                        if (newOpacity) {
+                            label.removeClass('highcharts-simulation-dataLabels');
+                            label.addClass('highcharts-simulation-dataLabels-fadeout');
```

Comment:
1. Do we need two classes, instead od one?
2. Classes should not be applied only when styled mode?